OpenSimRT/RealTime/tests/TestRTFromFile.cpp

```diff
@@ -0,0 +1,255 @@
+    muscleOptimizationParameters.maximumIterations = maximumIterations;
+    muscleOptimizationParameters.objectiveExponent = objectiveExponent;
+
+    // pipeline
+    RealTimeAnalysis::Parameters pipelineParameters;
+    pipelineParameters.solveMuscleOptimization = solveMuscleOptimization;
+    pipelineParameters.ikMarkerTasks = markerTasks;
+    pipelineParameters.ikConstraintsWeight = SimTK::Infinity;
+    pipelineParameters.ikAccuracy = 1e-5;
+    pipelineParameters.filterParameters = filterParameters;
+    pipelineParameters.muscleOptimizationParameters =
+            muscleOptimizationParameters;
+    pipelineParameters.wrenchParameters = wrenchParameters;
+    pipelineParameters.dataAcquisitionFunction = dataAcquisitionFunction;
+    pipelineParameters.momentArmFunction = calcMomentArm;
+    RealTimeAnalysis pipeline(model, pipelineParameters);
+    auto log = pipeline.initializeLoggers();
+
+    // run pipeline
+    pipeline.run();
+
+    // visualizer
+    BasicModelVisualizer visualizer(model);
+    auto rightGRFDecorator = new ForceDecorator(Green, 0.001, 3);
+    visualizer.addDecorationGenerator(rightGRFDecorator);
+    auto leftGRFDecorator = new ForceDecorator(Green, 0.001, 3);
+    visualizer.addDecorationGenerator(leftGRFDecorator);
+    auto rightKneeForceDecorator = new ForceDecorator(Red, 0.0005, 3);
+    visualizer.addDecorationGenerator(rightKneeForceDecorator);
+    auto leftKneeForceDecorator = new ForceDecorator(Red, 0.0005, 3);
+    visualizer.addDecorationGenerator(leftKneeForceDecorator);
+
+    try {
+        while (!pipeline.shouldTerminate()) {
+            // fetch of rt results
+            auto results = pipeline.getResults();
+
+            // update visualizer
+            if (!solveMuscleOptimization)
+                visualizer.update(results.q);
+            else {
+                visualizer.update(results.q, results.am);
+                visualizer.updateReactionForceDecorator(
+                        results.reactionWrenches, "tibia_r",
+                        rightKneeForceDecorator);
+                visualizer.updateReactionForceDecorator(
+                        results.reactionWrenches, "tibia_l",
+                        leftKneeForceDecorator);
+            }
+            // log
+            log.qLogger.appendRow(results.t, ~results.q);
+            log.qDotLogger.appendRow(results.t, ~results.qd);
+            log.qDDotLogger.appendRow(results.t, ~results.qdd);
+            log.tauLogger.appendRow(results.t, ~results.tau);
+            log.fmLogger.appendRow(results.t, ~results.fm);
+            log.amLogger.appendRow(results.t, ~results.am);
+            log.residualLogger.appendRow(results.t, ~results.residuals);
+            log.jrLogger.appendRow(results.t, ~results.reactionWrenchVector);
+        } // while loop
+    } catch (const exception& e) {
+        cout << e.what() << "\n";
+        pipeline.shouldTerminate(true);
+    }
+
+    // // store results
```

Comment:
Maybe we have a folder in real_time and we do the comparison to ensure everything works properly.

Reply:
Comparison of the results is not feasible. Due to the use of threads the method is not deterministic and the results are not consistent between runs. I tried it and it failed. This was my main concern regarding if these files should be tests or solely applications.

Reply:
Maybe a qualitative check in case the FPS is lower.

Reply:
Just to validate that we did not mess things. You could adopt a python script to plot and compare results.

Reply:
That would be a lot of plots...

Reply:
I tried to rework the `compareTables` function to compare rows (as an option) instead of columns only, in order to compare common row entries in the tables. However I noticed that the results of the modules in the processing thread depend on the input value history, and since the results are not passed to the processing thread in a deterministic manner, the RMSE often exceeds the given threshold.

Reply:
Results are placed in the buffer. It might be possible to lower the FPS of the acquisition thread so that the processing thread does not miss any frame. However, this depends on the machine being used for testing. We could have this as a test to verify that everything works but exclude the comparison afterward.